docs/design/architecture.md

```diff
@@ -170,47 +397,89 @@ The following diagram illustrates the Kata Containers network hotplug workflow.
+
+### Devicemapper
 
-Kata Containers has the ability to hotplug and remove block devices, which makes it possible to use block devices for containers started after the VM has been launched.
+The
+[devicemapper `snapshotter`](https://github.com/containerd/containerd/tree/master/snapshots/devmapper)
+is a special case. The `snapshotter` uses dedicated block devices
+rather than formatted filesystems, and operates at the block level
+rather than the file level. This knowledge is used to directly use the
+underlying block device instead of the overlay file system for the
+container root file system. The block device maps to the top
+read-write layer for the overlay. This approach gives much better I/O
+performance compared to using `virtio-fs` to share the container file
+system.
 
-Users can check to see if the container uses the devicemapper block device as its rootfs by calling `mount(8)` within the container.  If the devicemapper block device
-is used, `/` will be mounted on `/dev/vda`.  Users can disable direct mounting of the underlying block device through the runtime configuration.
+#### Hot plug and unplug
+
+Kata Containers has the ability to hot plug add and hot plug remove
+block devices. This makes it possible to use block devices for
+containers started after the VM has been launched.
+
+Users can check to see if the container uses the `devicemapper` block
+device as its rootfs by calling `mount(8)` within the container.  If
+the `devicemapper` block device is used, the root filesystem (`/`)
+will be mounted from `/dev/vda`.  Users can disable direct mounting of
+the underlying block device through the runtime
+[configuration](#configuration).
 
 ## Kubernetes support
 
-[Kubernetes\*](https://github.com/kubernetes/kubernetes/) is a popular open source
+[Kubernetes](https://github.com/kubernetes/kubernetes/), or K8s is a popular open source
 container orchestration engine. In Kubernetes, a set of containers sharing resources
 such as networking, storage, mount, PID, etc. is called a
-[Pod](https://kubernetes.io/docs/user-guide/pods/).
+[pod](https://kubernetes.io/docs/user-guide/pods/).
+
 A node can have multiple pods, but at a minimum, a node within a Kubernetes cluster
 only needs to run a container runtime and a container agent (called a
 [Kubelet](https://kubernetes.io/docs/admin/kubelet/)).
 
-A Kubernetes cluster runs a control plane where a scheduler (typically running on a
-dedicated master node) calls into a compute Kubelet. This Kubelet instance is
-responsible for managing the lifecycle of pods within the nodes and eventually relies
-on a container runtime to handle execution. The Kubelet architecture decouples
-lifecycle management from container execution through the dedicated
-`gRPC` based [Container Runtime Interface (CRI)](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/container-runtime-interface-v1.md).
+Kata Containers represents a Kubelet pod as a VM.
```

Comment:
Small nit but these are three individual sentences in a row each as their own paragraph.